src/utils/openwhisk.js

```diff
@@ -14,6 +14,19 @@
 const querystring = require('querystring');
 const owwrapper = require('@adobe/openwhisk-loggly-wrapper');
 
+/**
+ * Builds the URL from path, selector, extension and params
+ * @param {Object} p The request parameters
+ */
+function buildUrl(p) {
+  let path = p.path || '/';
+  if (path.indexOf('.') !== -1) path = path.substring(0, path.lastIndexOf('.'));
```

Comment:
no 1-liners for if. didn't eslint complain?

Reply:
don't calculate lastIndexOf() twice. store it in a variable first.

Reply:
also, I think the `path` here, is the path to the resource (i.e. the markdown file)

Reply:
I'd feel much better if we would use a URI parser here.

Reply:
as @trieloff mentioned below, the path already incudes the selector/extension. so no need to construct it here.

Reply:
> no 1-liners for if. didn't eslint complain?

no :) but I added a new block now

> don't calculate lastIndexOf() twice. store it in a variable first.

done

> the path already incudes the selector/extension. so no need to construct it here.

in my tests with [Adobe-Developer-Site](https://github.com/adobe/Adobe-Developer-Site), that isn't the case: `path` contains the path to the .md file, so we need to cut off the .md extension, add the `selector` and then `extension` from the `request.params`.

> I'd feel much better if we would use a URI parser here.

I can look into that next, however the current version behaves stably in my tests, with and without selector, even without extension. 

Examples:
```
{
  path: "/test.md",
  extension: "html",
  selector: "",
  query: "foo=bar"
}
```
returns `/test.html?foo=bar`, and
```
{
  path: "/test.md",
  selector: "dump",
  query: ""
}
```
returns `/test.dump.html`.

Reply:
> I'd feel much better if we would use a URI parser here.

@trieloff I looked into it and don't think a parser would be much help in this case, because we have to compute the path segment of the `url` based on the md path, the selector (if it exists) and the extension. i therefore doubt that the code would become much more readable than it is now. But of course I wouldn't mind being convinced otherwise.